src/twisted/conch/ssh/filetransfer.py

```diff
@@ -527,6 +534,32 @@ def connectionMade(self):
         self.sendPacket(FXP_INIT, data)
 
 
+    def connectionLost(self, reason):
+        """
+        Abort any pending requests.
+        """
+
+        # On Python2, we're an oldstyle class, so we can't use super().
+        FileTransferBase.connectionLost(self, reason)
+        # Not sure why self.connected isn't updated in
+        # t.i.p.Protocol.connectionLost() instead
+        self.connected = False
+
+        # If there are still requests waiting for responses when the
+        # connection is lost, fail them.
+        if self.openRequests:
+            if reason.check(error.ConnectionDone):
```

Comment:
why do we have this condition here.
my understanding is that this will always be called with ConnectionDone... so `else` is newer called.

Reply:
I'm also curious about this.

Reply:
Because  it seems like a bug that the `reason` is always `ConnectionDone` even in the case of a non-clean closure. Anyway, I've rewritten this not to special-case `ConnectionDone`.